### PR TITLE
Adding json logging for k8se apps

### LINKED ISSUE #7294: Change console logs to JSON format - in kubernetes environment
In kubernetes environment, system logs flow through the stdout stream of the pod, which gets sent to a shared log analytics workspace. The logs which get emitted to stdout needs to be in json format. Currently the [WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs) emits logs to stdout in a csv format. So we need to update the event generator to emit system logs in json format.

### PR DESCRIPTION
**Pull request by divyagandhisethi**
### Issue describing the changes in this PR

resolves #7294 

### Pull request checklist

* [X] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [X] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [X] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [X] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by divyagandhisethi** [comment]
cc @sanchitmehta

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs [hunk 1]
```diff
@@ -20,34 +22,62 @@ public KubernetesEventGenerator(IEnvironment environment, Action<string> writeEv
...
+            metricEvent.Add("EventName", eventName);
+            metricEvent.Add("Average", average);
+            metricEvent.Add("Minimum", minimum);
+            metricEvent.Add("Maximum", maximum);
+            metricEvent.Add("Count", count);
+            metricEvent.Add("HostVersion", ScriptHost.Version);
+            metricEvent.Add("EventTimeStamp", eventTimestamp.ToString(EventTimestampFormat));
+            metricEvent.Add("Data", NormalizeString(data));
+            metricEvent.Add("RuntimeSiteName", runtimeSiteName);
+            metricEvent.Add("SlotName", slotName);
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:67]: include podname here

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs [hunk 2]
```diff
@@ -20,34 +22,62 @@ public KubernetesEventGenerator(IEnvironment environment, Action<string> writeEv
...
 
-            _writeEvent($"{subscriptionId},{appName},{functionName},{eventName},{average},{minimum},{maximum},{count},{hostVersion},{eventTimestamp.ToString(EventTimestampFormat)},{NormalizeString(data)},{runtimeSiteName},{slotName},{_podName}");
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:35]: why are these being deleted?

**Review by pragnagopa — CHANGES_REQUESTED** [review]

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs (hunk 2 above)
Comment (divyagandhisethi) [reply src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:35]: It is a no-op today.

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs (hunk 1 above)
Comment (divyagandhisethi) [reply src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:67]: Pod name will be included by the logging agent.

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs (hunk 2 above)
Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:35]: Can you please expand on that? I see these methods are using `_writeEvent` as well - This PR is explicitly removing logs. Am I missing something here?

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs [hunk 3]
```diff
@@ -3,56 +3,83 @@
...
 
         public override void LogFunctionExecutionEvent(string executionId, string siteName, int concurrency, string functionName, string invocationId, string executionStage, long executionTimeSpan, bool success)
         {
-            _writeEvent($"{executionId},{siteName},{concurrency},{functionName},{invocationId},{executionStage},{executionTimeSpan},{success},{_podName}");
         }
 
         public override void LogAzureMonitorDiagnosticLogEvent(LogLevel level, string resourceId, string operationName, string category, string regionName, string properties)
         {
-            _writeEvent($"{(int)ToEventLevel(level)},{resourceId},{operationName},{category},{regionName},{NormalizeString(properties.Replace("'", string.Empty))},{DateTime.UtcNow.ToString()},{_podName}");
+            _writeEvent($"{(int)ToEventLevel(level)},{resourceId},{operationName},{category},{regionName},{NormalizeString(properties.Replace("'", string.Empty))},{DateTime.UtcNow.ToString()}");
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs]: Should this be JSON as well? as there is no schema in this environment?

Comment (divyagandhisethi) [reply src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs]: This is different from Kusto logs. These logs go to azure monitor, where a schema does need to be enforced.

**Comment by pragnagopa** [comment]
Tagging @cgillum  FYI

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs (hunk 2 above)
Comment (divyagandhisethi) [reply src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:35]: LogFunctionsDetailsEvent is not for Kusto logging today. Today we only use LogFunctionTraceEvent and LogFunctionsMetricEvent for Kusto logging. So we only need to send those events to the shared log analytics.

**Comment by cgillum** [comment]
Thanks @pragnagopa. It looks like we'll need to create a work item for Durable Functions to ensure we use JSON console logging when running on Kubernetes. We support JSON console logging, but it's designed only to work with Linux Consumption.

**Comment by fabiocav** [comment]
@divyagandhisethi is there more context on the requirements/details for this change? The issue doesn't have a description and I would like to understand the requirements and what this change is trying to accomplish before reviewing it.

**Comment by divyagandhisethi** [comment]
> 
> 
> @divyagandhisethi is there more context on the requirements/details for this change? The issue doesn't have a description and I would like to understand the requirements and what this change is trying to accomplish before reviewing it.

Thanks @fabiocav . I have updated the issue description. Let me know if you have any other questions.

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs [hunk 4]
```diff
@@ -3,56 +3,94 @@
...
         }
 
         public override void LogFunctionExecutionAggregateEvent(string siteName, string functionName, long executionTimeInMs, long functionStartedCount, long functionCompletedCount, long functionFailedCount)
         {
-            _writeEvent($"{siteName},{functionName},{executionTimeInMs},{functionStartedCount},{functionCompletedCount},{functionFailedCount},{_podName}");
         }
 
         public override void LogFunctionExecutionEvent(string executionId, string siteName, int concurrency, string functionName, string invocationId, string executionStage, long executionTimeSpan, bool success)
         {
-            _writeEvent($"{executionId},{siteName},{concurrency},{functionName},{invocationId},{executionStage},{executionTimeSpan},{success},{_podName}");
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:50]: Please add comments why this is a no-op when running in Kubernetes env - i.e. no state being saved. we rely on Keda scaler for scaling up and down

File: src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs [hunk 5]
```diff
@@ -3,56 +3,94 @@
...
+            traceLog.Add("SlotName", slotName);
+
+            _writeEvent(traceLog.ToString(Formatting.None));
         }
 
         public override void LogFunctionMetricEvent(string subscriptionId, string appName, string functionName, string eventName, long average, long minimum, long maximum, long count, DateTime eventTimestamp, string data, string runtimeSiteName, string slotName)
         {
-            string hostVersion = ScriptHost.Version;
+            JObject metricEvent = new JObject();
+            metricEvent.Add("EventType", ScriptConstants.LinuxLogEventStreamName);
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs:54]: Please add comments - how these prefixes will used when host is running in Kubenetes env.

**Review by pragnagopa — APPROVED** [review]
👍

**Pull request merged**